blog/content/post/2025-10-24-crosscon_cba.md

```diff
@@ -0,0 +1,344 @@
+like reflections or scattering. All that information creates a distinct
+fingerprint (or context) that identifies the device's environment[^csi-inf].
+
+### How "true" is the context?
+
+I personally could not comprehend two questions regarding CBA:
+
+* Why common identifiers like MAC addresses are not part of the fingerprint?
+* Doesn't the "resolution" of the received information take a huge role in
+  the level of detail of the captured information?
+
+The answer to the first question is quite obvious. The identifiers, like MAC or
+IPs, are easy to spoof. While in the context of huge datacenters an operation
+like that might be hard to hard to perform, it is a valid threat for less dense
+environments. Another fact is that classic networking, meaning digital
+processing, is computation-heavy as it relies on cryptographic operations, which
+is a concern often constrained IoT devices. The environmental data in the form
+of physical electromagnetic signals is both harder to replicate (spoof) and
+requires less computational overhead, as it does not need to be processed
+"on-board".
+
+For the second question, the resolution, consistency, and robustness of the
+captured data take a huge role. The low-end receiver's capabilities to capture
+subtle differences are not as good as those of high-end devices. Moreover, the
+hardware can age, and the results will drift over time. While the machine
+learning algorithms don’t compensate for low-resolution output, they can “learn”
+the distinct features within the CSI data that remain discriminative across
+different devices.
+
+### Particular CBA implementation
+
+_Note: below, I'll be describing the particular implementation that we received.
+The implementation is at the proof-of-concept stage._
+
+The Context-Based Authentication implementation we received uses the onboard
+Wi-Fi chip to gather CSI (Channel State Information) of the nearby devices, just
+as described earlier. The host device used as a reference platform is the
+Raspberry Pi 4. The onboard WiFi chip takes the role of a sniffer in an attempt
+to capture various transmissions that happen over the channel, .eg, beacon and
+probe frames. The output of this process is a set of CSI information,
+information about physical channel properties. All the information the device
+manages to gather is what is referred to as a fingerprint or a context. The
+fingerprint will be slightly different each time the probing is performed, and
+that is to be expected. This is because the devices can change location, or the
+signals themselves might get distorted. The authentication scheme assumes there
+is another entity, an authority that assesses the match rate (similarity)
+between enrollment and verification. If the fingerprint created during
+assessment has a sufficient match rate, then the authority grants the signature
+to the assessing node. That signature can later be used to authenticate to yet
+another service.
```

Comment:
This paragraph is dense; it would benefit from a sequence diagram, a picture, graphics, or any concept visualization.